src/cpp/shared_core/system/User.cpp

```diff
@@ -161,6 +166,48 @@ FilePath User::getUserHomePath(const std::string& in_envOverride)
    return FilePath(getEnvVariable("HOME"));
 }
 
+User& User::operator=(const User& in_other)
+{
+   m_impl->Name = in_other.m_impl->Name;
```

Comment:
Is it intentional that we explicitly copy the members piece-by-piece rather than just relying on `*m_impl = *in_other->m_impl` or something similar? (Main worry is that it could be easy to forget to update this method if we were to add new members to the Impl struct)

Reply:
I _think_ that should be safe. I was probably just being overly untrusting of the compiler generated code at the time. I'll make that change.